velox/expression/SignatureBinder.cpp

```diff
@@ -33,56 +33,52 @@ bool isPositiveInteger(const std::string& str) {
       }) == str.end();
 }
 
-std::string buildCalculation(
-    const std::string& variable,
-    const std::string& calculation) {
-  return fmt::format("{}={}", variable, calculation);
-}
-
 TypePtr inferDecimalType(
     const exec::TypeSignature& typeSignature,
-    std::unordered_map<std::string, int>& variables,
+    const std::unordered_map<std::string, std::optional<int>>&
+        integerParameters,
     const std::unordered_map<std::string, std::string>& constraints) {
-  const auto& precisionVar = typeSignature.variables()[0];
-  const auto& scaleVar = typeSignature.variables()[1];
+  const auto& precisionVar = typeSignature.parameters()[0].baseName();
+  const auto& scaleVar = typeSignature.parameters()[1].baseName();
   int precision = 0;
   int scale = 0;
   // Determine precision.
   if (isPositiveInteger(precisionVar)) {
     // Handle constant.
     precision = atoi(precisionVar.c_str());
-  } else if (variables.find(precisionVar) != variables.end()) {
+  } else if (
+      (integerParameters.count(precisionVar) > 0) &&
+      integerParameters.find(precisionVar)->second.has_value()) {
     // Check if variable is already computed.
-    precision = variables[precisionVar];
+    precision = integerParameters.find(precisionVar)->second.value();
```

Comment:
nit: Could we avoid the recomputation of `integerParameters.find(precisionVar)`?